### PR TITLE
Test 95

### PR DESCRIPTION
**Pull request by kristiyanplpeev**
Front end added select buttons for selecting which team and position i want to be displayed, also changed a bit service method for getting all users.

Back end changed method for getting users by adding filtering functionality.

### CODE REVIEW

File: client/src/components/UsersList/UsersList.tsx [hunk 1]
```diff
@@ -91,11 +168,11 @@ class UsersList extends Component<UsersListProps, UsersListState> {
     );
   }
 
-  renderUsers(): Array<JSX.Element> {
+  renderUsers(): Array<JSX.Element> | JSX.Element {
```
Comment (MilenIvanovv) [inline client/src/components/UsersList/UsersList.tsx:171]: Omit Array<JSX.Element> |
Make it to return 
(
    <>
    this.state.users...
    </>
)

File: client/src/services/UserService.ts [hunk 2]
```diff
@@ -19,8 +19,9 @@ class UserService implements IUserService {
     Authorization: `Bearer ${this.authService.getToken()}`,
   };
 
-  getAllUsers = async (): Promise<Array<IUserWithTeamAndPosition>> => {
-    return (await axios.get(`${BASE_URL}users`, { headers: this.headers })).data;
+  getAllUsers = async (teamId: string, positionId: string): Promise<Array<IUserWithTeamAndPosition>> => {
+    return (await axios.get(`${BASE_URL}users?teamId=${teamId}&positionId=${positionId}`, { headers: this.headers }))
```
Comment (MilenIvanovv) [inline client/src/services/UserService.ts:23]: Ussualy when there is a query param for filtering, like here, they are optional. When you don't provide them the request should return every user.

File: server/src/users/users.controller.ts [hunk 3]
```diff
@@ -13,8 +13,11 @@ export class UsersController {
 
   @Get()
   @UseGuards(JwtAuthGuard)
-  public async getUsers(): Promise<Array<UserDetailsWithTeamAndPosition>> {
-    return await this.usersService.getAllUsers();
+  public async getUsers(
+    @Query('teamId') teamId: string,
+    @Query('positionId') positionId: string,
```
Comment (MilenIvanovv) [inline server/src/users/users.controller.ts:18]: If choose leave the query params required, it is a good practice to check if they are undefined and throw badrequest

**Comment by MilenIvanovv** [comment]
LGTM, left some comments for minor changes

**Review by MilenIvanovv — CHANGES_REQUESTED** [review]
LGTM, left some comments for minor changes

File: client/src/components/UsersList/UsersList.tsx (hunk 1 above)
Comment (kristiyanplpeev) [reply client/src/components/UsersList/UsersList.tsx:171]: done.

File: client/src/services/UserService.ts (hunk 2 above)
Comment (kristiyanplpeev) [reply client/src/services/UserService.ts]: done.

File: server/src/users/users.controller.ts (hunk 3 above)
Comment (kristiyanplpeev) [reply server/src/users/users.controller.ts:18]: done.

**Review by MilenIvanovv — APPROVED** [review]

**Pull request merged**